third_party/terraform/utils/import.go

```diff
@@ -25,7 +26,37 @@ func parseImportId(idRegexes []string, d TerraformResourceData, config *Config)
 			// Starting at index 1, the first match is the full string.
 			for i := 1; i < len(fieldValues); i++ {
 				fieldName := re.SubexpNames()[i]
-				d.Set(fieldName, fieldValues[i])
+				// This part looks confusing.  Because there is no way to know at
+				// this point whether 'fieldName' corresponds to a TypeString or a
+				// TypeInteger in the resource schema, we need to determine
+				// whether to call d.Set() with 'fieldValues[i]', or with an integer
+				// parsed from 'fieldValues[i]'.  Normally, we would be able to just
+				// use a try/catch pattern - try as a string, and if that doesn't
+				// work, try as an integer, and if that doesn't work, return the
+				// error.  Unfortunately, this is not possible here - during tests,
+				// d.Set(...) will panic if there is an error.  So we try d.Get(),
+				// relying on the behavior that an unset field will return the empty
+				// value for the appropriate type.  So if the result is a number, we'll
+				// set it as a number, but if it's a string, we'll set it as a string.
+				val, _ := d.GetOk(fieldName)
+				if _, ok := val.(string); ok {
+					if err = d.Set(fieldName, fieldValues[i]); err != nil {
+						return err
+					}
+				} else if _, ok := val.(int); ok {
+					if atoi, atoiErr := strconv.Atoi(fieldValues[i]); atoiErr == nil {
```

Comment:
I have a slight preference for [`ParseInt`](https://golang.org/pkg/strconv/#ParseInt) over atoi, mostly just because the name of `ParseInt` is more clear than `Atoi` imo.

Reply:
`Atoi` is "`A`scii `to` `i`nteger" - it's the name for this function in C, and aesthetically I prefer it because having '10' and '64' on the line (as is necessary for `ParseInt`) looks nasty (plus, `atoi` returns `int`, which is what you need in the `d.Set` call, where `ParseInt` returns `int64`, so there's an extra cast).  Here's the change, but which way looks better to you?

```diff
diff --git a/third_party/terraform/utils/import.go b/third_party/terraform/utils/import.go
index ff86d9fa..2b158b7c 100644
--- a/third_party/terraform/utils/import.go
+++ b/third_party/terraform/utils/import.go
@@ -44,10 +44,10 @@ func parseImportId(idRegexes []string, d TerraformResourceData, config *Config)
                                                return err
                                        }
                                } else if _, ok := val.(int); ok {
-                                       if atoi, atoiErr := strconv.Atoi(fieldValues[i]); atoiErr == nil {
+                                       if intVal, parseErr := strconv.ParseInt(fieldValues[i], 10, 64); parseErr == nil {
                                                // If the value can be parsed as an integer, we try to set the
                                                // value as an integer.
-                                               if err = d.Set(fieldName, atoi); err != nil {
+                                               if err = d.Set(fieldName, int(intVal)); err != nil {
                                                        return err
                                                }
                                        } else {
```

Reply:
I think we can do a `d.Set` with an `int64` and Terraform does the conversion for us, and the `64` *could* be a `0` (since `Atoi` makes the call `ParseInt(s, 10, 0)` afaik). If `Atoi` is your preference here feel free, this is just aesthetics.

Reply:
I do prefer it, I'm gonna leave it if it's not too much a problem.  Happy to swap it if your preference is stronger than mine, resolving, but let me know.